File: golang/122_Best_Time_to_buy_and_sell_stock_II.go
func maxProfit(prices []int) int {
    b, e, maxprofit := 0, 1, 0
    for ; e < len(prices); {
        if (prices[e] > prices[b]){
            maxprofit += prices[e] - prices[b]
        }
        b, e = e, e+1
    }
    return maxprofit
}

File: leetcode/Python/122_Best_Time_to_Buy_and_Sell_Stock_II.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        b, e, max_profit = 0, 1, 0
        while e < len(prices):
            if prices[e]-prices[b] > 0:
                max_profit += prices[e]-prices[b]
            b, e = e, e+1
        return max_profit